OMCompiler/SimulationRuntime/c/util/read_matlab4.c

```diff
@@ -141,17 +141,20 @@ void omc_free_matlab4_reader(ModelicaMatReader *reader)
   }
 }
 
-void remSpaces(char *ch){
-    char *ch2 = ch;
-    unsigned int ui = 0;
-    unsigned int uj = 0;
-
-    for(ui=0;ui<=strlen(ch);ui++){
-        if(ch[ui]!=' '){
-            ch2[uj] = ch[ui];
-            uj++;
-        }
-    }
+void trimWhitespace(char *str)
+{
+  char *end;
+  /* skip leading whitespace */
+  while (isspace(*str)) {
+    str++;
+  }
```

Comment:
This just advances the local `str` pointer, it doesn't change the string. If the intent is to remove leading whitespace you need to actually change the string like the original function.

Reply:
```C
void trimWhitespace(char *str)
{
  char *start = str;
  char *end;
  size_t len;

  /* find first non-whitespace */
  while (isspace(*start)) {
    ++start;
  }

  /* find last non-whitespace */
  end = start + strlen(start) - 1;
  while (end > str && isspace(*end)) {
    --end;
  }

  len = end-start+1;

  /* shift the string to remove leading whitespace */
  if (start != str) memmove(str, start, len);

  /* terminate the string to remove trailing whitespace */
  *(str+len) = '\0';
}
```

Reply:
Yes, all the reference files variables only have the trailing spaces so I forgot to remove the leading spaces :smile:. I updated and improved the code.